bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -553,4 +573,32 @@
 		<description>Name of the scenario to trigger</description>
 	</channel-type>
 
+	<channel-type id="bypass-state">
+		<item-type>String</item-type>
+		<label>Bypass State</label>
+		<description>Indicates whether the device is bypassed.</description>
+		<state readOnly="true">
+			<options>
+				<option value="BYPASS_INACTIVE">Bypass is not active</option>
+				<option value="BYPASS_ACTIVE">Bypass is active</option>
+				<option value="UNKNOWN">Bypass state cannot be determined</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="communication-quality">
+		<item-type>String</item-type>
+		<label>Communication Quality</label>
+		<description>The quality of communication between a device and the Bosch Smart Home Controller.</description>
+		<state readOnly="true">
+			<options>
+				<option value="BAD">Bad</option>
+				<option value="MEDIUM">Medium</option>
+				<option value="NORMAL">Normal</option>
+				<option value="GOOD">Good</option>
+				<option value="UNKNOWN">Unknown</option>
+			</options>
+		</state>
+	</channel-type>
+
```

Comment:
This could probably be mapped into system channel type `signal-strength`:
https://github.com/openhab/openhab-core/blob/369bd228e7a361b1c884b9af5129304291243f91/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/DefaultSystemChannelTypeProvider.java#L109-L123

0. no signal - unknown?
1. weak = bad
2. average = normal
3. good = good
4. excellent = no corresponding value (not used)

This way you would benefit from the unified vendor-agnostic representation which also includes icons representing the signal quality.

Reply:
Thanks for the hint, I will enhance the code as soon I find some time again :+1:

Reply:
I enhanced the code using all 5 possible states:

0. `UNKNOWN` -> 0
1. `BAD` -> 1
2. `MEDIUM` -> 2
3. `NORMAL` -> 3
4. `GOOD` -> 4

Does that make sense?

Reply:
Yes, it makes perfect sense. 👍